### ISSUE TITLE
Shard metrics collector for throttler should be able to config TLS or disabled

### ISSUE THREAD

**Issue by sonne5** [vitessio/vitess#9213]
#### Feature Description
When we enable throttler on VTTablet, it automatically enables both `Shard` and `Self`. For self, it just use DBA connection pool to query mysql metrics. However, with `Shard` build connection via formatted string: https://github.com/vitessio/vitess/blob/main/go/vt/vttablet/tabletserver/throttle/mysql/probe.go#L85

This is problematic if mysql is bring up with `--require_secure_transport=ON`. As a result:
```
$ curl -s http://localhost:19000/throttler/status | jq .
{
  "Keyspace": "sbtest",
  "Shard": "0",
  "IsLeader": true,
  "IsOpen": true,
  "IsDormant": false,
  "AggregatedMetrics": {
    "mysql/self": {
      "Value": 0.456644
    },
    "mysql/shard": {
      "ClusterName": "shard",
      "Key": {
        "Hostname": "xxxxx",
        "Port": 17000
      },
      "Value": 0,
      "Err": {
        "Number": 3159,
        "Message": "Connections using insecure transport are prohibited while --require_secure_transport=ON."
      }
    }
  },
  "MetricsHealth": {
    "mysql/self": {
      "LastHealthyAt": "2021-11-12T17:04:17.018064082Z",
      "SecondsSinceLastHealthy": 0
    }
  }
```

I think we should either:
- allow adding `tls` in the url connection
- allow disable shard throttler check, especially if `throttle_check_as_check_self` is set

#### Use Case(s)
Any one that's using throttler with `--require_secure_transport=ON` might run into this issue

**Comment by sonne5** [comment]
@aquarapid Could you help and take a look to see if this feature makes sense? If so, mind helping triage this one? Thanks!

**Comment by mattlord** [comment]
Hi @sonne5,

Thank you for the helpful issue! I see a bigger underlying problem here around how these connections are managed and we're discussing that internally. The issue you mentioned here is simply 1 of many potential problems one could encounter.

But in this specific case, I would expect using `-throttle_check_as_check_self` would effectively disable the shard level check (https://vitess.io/docs/reference/features/tablet-throttler/#configuration):
> Use -throttle_check_as_check_self to implicitly reroute any /throttler/check call into /throttler/check-self. This makes sense when the user supplies a custom query, and where the user wishes to throttle writes to the cluster based on the primary tablet’s health, rather than the overall health of the cluster.

I have confirmed this behavior:
```
vitess@94aad2283aff:/vt/local$ curl -s localhost:15100/throttler/check
{"StatusCode":200,"Value":0.069765,"Threshold":1,"Message":""}
vitess@94aad2283aff:/vt/local$ curl -s localhost:15100/throttler/check-self
{"StatusCode":200,"Value":0.153103,"Threshold":1,"Message":""}
```

That is the behavior you want here, right? Or perhaps I misunderstood.

Thanks again!

**Comment by sonne5** [comment]
@mattlord thanks for looking into this. 
Yes, the `-throttle_check_as_check_self` is the behavior we want, the problem is enabling that flag does not prevent vitess sending "shard" query for probing... the sql nodes quickly run into "too many connection" errors because we don't cached "failed" connection and reuse later: https://github.com/vitessio/vitess/blob/a9289e01471560e15359bdd355f01dd31ae82d5c/go/vt/orchestrator/external/golib/sqlutils/sqlutils.go#L184-L188

Let me know if I'm missing something here, thanks!

**Comment by mattlord** [comment]
@sonne5 you're right, the flag only [affects the handling of HTTP interface responses](https://github.com/vitessio/vitess/blob/f0e9966fcb589cfa12e10883c8d6286091b24251/go/vt/vttablet/tabletserver/throttle/throttler.go#L839-L843) — it does not affect the underlying collection (polling) and caching of the metrics that are served via the HTTP interface.

This is a bug and we'll spend some time on a PR. Thanks again!

**Comment by shlomi-noach** [comment]
Addressed by #9334

**Comment by mattlord** [comment]
Hi @sonne5,

This was resolved by leveraging the existing HTTP interface when gathering the Shard level metrics from remote tablets — so hitting `/throttler/check-self` on remote tablets and no longer connecting directly to remote tablet mysqld instances. I'm closing this as fixed for now, but please let me know if I'm missing or misunderstanding something and we can re-open.

Thank you for the bug report!
